src/Toml/Printer.hs

```diff
@@ -167,3 +172,26 @@ prettyTables options i pref =
 
 prettyKey :: Key -> Text
 prettyKey (Key k) = Text.intercalate "." $ map unPiece (NonEmpty.toList k)
+
+prettyTableArrays :: PrintOptions -> Int -> Text -> HashMap Key (NonEmpty TOML) -> Text
+prettyTableArrays options i pref =
+    Text.intercalate "\n" . map arrText . orderWith options . HashMap.toList
+  where
+    arrText :: (Key, NonEmpty TOML) -> Text
+    arrText (k, ne) =
+      let name = getPref k
+          render toml = mconcat
+            [ tabWith options i
+            , prettyTableArrayName name
+            , prettyTomlInd options (succ i) name toml
+            ]
+      in Text.intercalate "\n" $ map render $ NonEmpty.toList ne
+
+    -- Adds next part of the table name to the accumulator.
+    getPref :: Key -> Text
+    getPref k = case pref of
+        "" -> prettyKey k
+        _  -> pref <> "." <> prettyKey k
```

Comment:
The same is in tables. Probably makes sense to move into separate function as well to remove code duplication.